invokeai/app/api/routers/models.py

```diff
@@ -90,23 +49,28 @@ async def list_models(
     return models
 
 @models_router.post(
-    "/",
+    "/{base_model}/{model_type}/{model_name}",
     operation_id="update_model",
     responses={200: {"status": "success"}},
 )
 async def update_model(
-    model_request: CreateModelRequest
+        base_model: BaseModelType = Path(default='sd-1', description="Base model"),
+        model_type: ModelType = Path(default='main', description="The type of model"),
+        model_name: str = Path(default=None, description="model name"),
+        info: Union[tuple(MODEL_CONFIGS)]  = Body(description="Model configuration"),
 ) -> CreateModelResponse:
     """ Add Model """
-    model_request_info = model_request.info
-    info_dict = model_request_info.dict()
-    model_response = CreateModelResponse(name=model_request.name, info=model_request.info, status="success")
-
     ApiDependencies.invoker.services.model_manager.add_model(
-        model_name=model_request.name,
-        model_attributes=info_dict,
+        model_name=model_name,
+        base_model=base_model,
+        model_type=model_type,
+        model_attributes=info.dict(),
```

Comment:
Per the PR description, only some of these attributes are actually changed. The `body` of the request should be typed to only include the changeable model attributes, it's very confusing if e.g. you are allowed to provide a new name, but it doesn't change anything

Reply:
The schema has now been fixed so that only changeable values are provided:
```
{
  "path": "string",
  "description": "string",
  "model_format": "checkpoint",
  "error": "not_found",
  "vae": "string",
  "config": "string",
  "variant": "normal"
}
```